### ISSUE TITLE
PDAL Exception export raster cloud

### ISSUE THREAD

**Issue by yotopo** [CloudCompare/CloudCompare#1622]
**Describe the bug**
after building a new cloud from  Tools -> Projections -> Rasterize, the export failed with the error : `PDAL Exception : Can't create dimension with invalid name " Height grid values"`

**How to reproduce**

Steps to reproduce the behaviour
1.Import .las
2. Go to `Tools -> Projections -> rasterize -> active layer height grid value ( but same error with others too) -> interpolate on Z`
3. Click on 'update grid -> Cloud then OK'
4. Click on `save` choose `LAS cloud`
5. Tick ` save additional field` ( or not the result is the same ) then `ok`
6. See the first error :  `PDAL Exception : Can't create dimension with invalid name " Height grid values"`
7. See the second error : `an error occured while saving /home/.../yourfile-Cloud : the third-party library in charge of saving/loading the file has thrown an exception`

**Your environment**
 - Archlinux
 - pdal 2.3.0-5
 - Cloud Compare version 2.12.0 but same error on 2.11

**Comment by dgirardeau** [comment]
I haven't been able to reproduce it on my side... Is it a Linux-only issue maybe?

Probably something to do with the space characters in the SF name... Can you try by manually replacing first the space characters by underscore for instance? (use F2 or 'Edit > Scalar fields > Rename' with this 'Height grid values' SF active).

**Comment by kikislater** [comment]
Hello,

Manjaro as well but running same issue with a flatpak package : 

Console output => 

```
[11:03:46] [Global Shift] Max abs. coord = 1e+04 / max abs. diag = 1e+06
[11:03:46] [ccColorScalesManager] Found 0 custom scale(s) in persistent settings
[11:03:46] [Plugin] Searching: /app/lib/cloudcompare/plugins
[11:03:46] 	Plugin found: Additional I/O (libQADDITIONAL_IO_PLUGIN.so)
[11:03:46] 	Plugin found: Animation (libQANIMATION_PLUGIN.so)
[11:03:46] 	Plugin found: CEA Virtual Broom (libQBROOM_PLUGIN.so)
[11:03:46] 	Plugin found: CANUPO (libQCANUPO_PLUGIN.so)
[11:03:46] 	Plugin found: Compass (libQCOMPASS_PLUGIN.so)
[11:03:46] 	Plugin found: Core I/O (libQCORE_IO_PLUGIN.so)
[11:03:46] 	Plugin found: Cork (libQCORK_PLUGIN.so)
[11:03:46] 	Plugin found: CSF Filter (libQCSF_PLUGIN.so)
[11:03:46] 	Plugin found: CSV Matrix I/O (libQCSV_MATRIX_IO_PLUGIN.so)
[11:03:46] 	Plugin found: E57 (libQE57_IO_PLUGIN.so)
[11:03:46] 	Plugin found: EDL Shader (libQEDL_GL_PLUGIN.so)
[11:03:46] 	Plugin found: Facet/fracture detection (libQFACETS_PLUGIN.so)
[11:03:46] 	Plugin found: Hough Normals Computation (libQHOUGH_NORMALS_PLUGIN.so)
[11:03:46] 	Plugin found: Hidden Point Removal (libQHPR_PLUGIN.so)
[11:03:46] 	Plugin found: M3C2 Distance (libQM3C2_PLUGIN.so)
[11:03:46] 	Plugin found: PCD file I/O (libQPCL_IO_PLUGIN.so)
[11:03:46] 	Plugin found: PCL wrapper (libQPCL_PLUGIN.so)
[11:03:46] 	Plugin found: PCV / ShadeVis (libQPCV_PLUGIN.so)
[11:03:46] 	Plugin found: PDAL (libQPDAL_IO_PLUGIN.so)
[11:03:46] 	Plugin found: Photoscan I/O (libQPHOTOSCAN_IO_PLUGIN.so)
[11:03:46] 	Plugin found: PoissonRecon (libQPOISSON_RECON_PLUGIN.so)
[11:03:46] 	Plugin found: RANSAC Shape Detection (libQRANSAC_SD_PLUGIN.so)
[11:03:46] 	Plugin found: RDB2 (I/O Plugin) (libQRDB_IO_PLUGIN.so)
[11:03:46] 	Plugin found: Surface of Revolution Analysis (libQSRA_PLUGIN.so)
[11:03:46] 	Plugin found: SSAO Shader (libQSSAO_GL_PLUGIN.so)
[11:03:46] [Plugin] Searching: /home/sylvain/.var/app/org.cloudcompare.CloudCompare/data/CCCorp/CloudCompare/plugins
[11:03:46] [Plugin] Searching: /app/share/CCCorp/CloudCompare/plugins
[11:03:46] [Plugin] Searching: /usr/share/CCCorp/CloudCompare/plugins
[11:03:46] [Plugin] Searching: /usr/share/runtime/share/CCCorp/CloudCompare/plugins
[11:03:46] [Plugin] Searching: /run/host/user-share/CCCorp/CloudCompare/plugins
[11:03:46] [Plugin] Searching: /run/host/share/CCCorp/CloudCompare/plugins
[11:03:46] [Plugin][Additional I/O] New file extensions registered: ICM OUT PN POLY POV PV SOI SX
[11:03:46] [Plugin][Core I/O] New file extensions registered:  GEOREF MA OBJ OFF PDMS PTX SBF STL VTK
[11:03:46] [Plugin][CSV Matrix I/O] New file extensions registered: CSV
[11:03:46] [Plugin][E57] New file extensions registered: E57
[11:03:46] [Plugin][PCD file I/O] New file extensions registered: PCD
[11:03:46] [Plugin][PDAL] New file extensions registered: LAS
[11:03:46] [Plugin][Photoscan I/O] New file extensions registered: PSZ
[11:03:46] [Plugin][RDB2 (I/O Plugin)] New file extensions registered: RDB2
[11:03:46] [ccGLWindow] Perspective is off by default
[11:03:46] CloudCompare started!
[11:03:46] [3D View 1] Graphics card manufacturer: Intel
[11:03:46] [3D View 1] Renderer: Mesa Intel(R) UHD Graphics 620 (KBL GT2)
[11:03:46] [3D View 1] GL version: 4.6 (Compatibility Profile) Mesa 21.3.8 (git-813ee839be)
[11:03:46] [3D View 1] GLSL Version: 4.60
[11:03:46] [3D View 1] VBOs available
[11:03:46] [3D View 1] Shaders available
[11:03:46] [3D View 1] GL filters available
[11:03:46] [3D View 1] Color ramp shader loaded successfully
[11:03:46] [3D View 1] Stereo mode: not supported
[11:03:46] [ccGLWindow] 3D view initialized
[11:03:46] [ccGLWindow] Succesfully initialized PBO for faster depth picking
[11:03:46] [Global Shift] Max abs. coord = 1e+04 / max abs. diag = 1e+06
[11:08:34] [LAS] Point format: 3
[11:08:36] [LAS] Spatial reference: LOCAL_CS["RGF93 / CC45",UNIT["metre",1,AUTHORITY["EPSG","9001"]],AXIS["Easting",EAST],AXIS["Northing",NORTH],AUTHORITY["EPSG","3945"]]
[11:08:37] [LAS] Cloud has been recentered! Translation: (-1940700.00 ; -4209300.00 ; 0.00)
[11:08:37] [LAS] Color components are coded on 16 bits
[11:08:37] [LAS] Time SF has been shifted to prevent a loss of accuracy (311235515.14)
[11:09:46] [LAS] All 'PointSourceId' values were the same (0)! We ignored them...
[11:09:46] [LAS] All 'UserData' values were the same (0)! We ignored them...
[11:09:46] [LAS] All 'EdgeOfFlightLine' values were the same (0)! We ignored them...
[11:09:46] [LAS] All 'ScanDirectionFlag' values were the same (0)! We ignored them...
[11:09:46] [LAS] All 'NumberOfReturns' values were the same (0)! We ignored them...
[11:09:46] [LAS] All 'ReturnNumber' values were the same (0)! We ignored them...
[11:09:48] [I/O] File '/home/sylvain/SIG/GEOBASE/lidar/ground.laz' loaded successfully
[11:10:04] [LoD] Preparing LoD acceleration structure for cloud 'ground - Cloud' [44996280 points]...
[11:10:14] [LoD] Level 0: 1 cells
[11:10:15] [LoD] Level 1: 8 cells
[11:10:16] [LoD] Level 2: 29 cells
[11:10:17] [LoD] Level 3: 66 cells
[11:10:18] [LoD] Level 4: 198 cells
[11:10:19] [LoD] Level 5: 680 cells
[11:10:20] [LoD] Level 6: 2469 cells
[11:10:21] [LoD] Level 7: 9025 cells
[11:10:22] [LoD] Level 8: 41865 cells
[11:10:23] [LoD] Level 9: 99737 cells
[11:10:23] [LoD] Level 10: 232010 cells
[11:10:24] [LoD] Level 11: 232794 cells
[11:10:24] [LoD] Level 12: 25291 cells
[11:10:24] [LoD] Level 13: 128 cells
[11:10:24] [LoD][pass 2] Level 1: 8 cells (+0)
[11:10:24] [LoD][pass 2] Level 2: 29 cells (+0)
[11:10:24] [LoD][pass 2] Level 3: 76 cells (+10)
[11:10:24] [LoD][pass 2] Level 4: 214 cells (+16)
[11:10:24] [LoD][pass 2] Level 5: 708 cells (+28)
[11:10:24] [LoD][pass 2] Level 6: 2524 cells (+55)
[11:10:24] [LoD][pass 2] Level 7: 9372 cells (+347)
[11:10:24] [LoD][pass 2] Level 8: 45208 cells (+3343)
[11:10:24] [LoD][pass 2] Level 9: 146385 cells (+46648)
[11:10:24] [LoD][pass 2] Level 10: 467886 cells (+235876)
[11:10:24] [LoD] Acceleration structure ready for cloud 'ground - Cloud' (max level: 13 / mem. = 56.80 Mb / duration: 20.1 s.)
[11:12:14] [ccGLWindow] Succesfully initialized PBO for faster depth picking
[11:12:39] [Rasterize] Current raster grid:
	Size: 854 x 1003
	Height values: [774.15 ; 789.5]
[11:12:39] Default point size is already at maximum : 16
[11:12:51] [Rasterize] Previously selected entity (source cloud) has been hidden!
[11:12:51] [Rasterize] Cloud 'ground-Cloud.raster(1)' successfully exported
[11:15:28] PDAL exception: Can't create dimension with invalid name 'Height grid values'.
[11:15:38] An error occurred while saving '/home/sylvain/SIG/GEOBASE/lidar/ground_rasterize': the third-party library in charge of saving/loading the file has thrown an exception
```

![image](https://user-images.githubusercontent.com/13238064/162166622-54cedba9-d67f-4bf4-9375-6bb1b0f901ef.png)


![image](https://user-images.githubusercontent.com/13238064/162166707-0333686a-6c29-409b-a7dd-e062070a09de.png)

**Comment by dgirardeau** [comment]
Same question, what happens if you rename this scalar field?

**Comment by kikislater** [comment]
Ah sorry, you mean renaming `Height grid values` to `Heightgridvalues`
Yes it works !

**Comment by dgirardeau** [comment]
Yep, you would have to do that for all scalar fields

**Comment by dgirardeau** [comment]
This should now be fixed in the latest 2.12.2 version!
